### ISSUE TITLE
When I'm in the terminal's accessibility buffer and I want to move focus to the terminal, I need to press the tab key twice

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#173780]
Type: <b>Bug</b>
CC @meganrogge 
CC @isidorn 

Using orca I open a terminal in vscode, type some commands, press shift+tab and explore the result using the arrow keys. After that I need to press the tab key twice to move the focus to the terminal. Previously I only had to press the tab key once.

VS Code version: Code - Insiders 1.76.0-insider (a7d49b0f35f50e460835a55d20a00a735d1665a3, 2023-02-08T05:24:48.579Z)
OS version: Linux x64 6.1.9-200.fc37.x86_64
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz (8 x 2999)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|15.27GB (8.27GB free)|
|Process Argv|--crash-reporter-id bc02adbd-6c3a-44b8-af92-55103a608af7|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (37)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|1.4.0
vscode-markdownlint|Dav|0.49.0
vscode-chatgpt|gen|3.5.6
rest-client|hum|0.25.1
better-cpp-syntax|jef|1.17.2
azure-pipelines|ms-|1.208.0
vscode-docker|ms-|1.23.3
csharp|ms-|1.25.4
vscode-kubernetes-tools|ms-|1.3.11
remote-containers|ms-|0.278.0
remote-ssh|ms-|0.97.2023020215
remote-ssh-edit|ms-|0.84.0
azure-account|ms-|0.11.3
cmake-tools|ms-|1.13.45
cpptools|ms-|1.14.1
cpptools-extension-pack|ms-|1.3.0
makefile-tools|ms-|0.6.0
powershell-preview|ms-|2023.2.0
remote-explorer|ms-|0.3.2023020609
vsliveshare|ms-|1.0.5828
fabric8-analytics|red|0.3.6
java|red|1.15.2023020803
vscode-xml|red|0.25.2023020103
vscode-yaml|red|1.11.0
sonarlint-vscode|Son|3.14.0
cmake|twx|0.0.17
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
vscode-boot-dev-pack|vmw|0.2.1
vscode-spring-boot|vmw|1.44.0
vscode-java-debug|vsc|0.48.0
vscode-java-dependency|vsc|0.21.2023013100
vscode-java-pack|vsc|0.25.2022112900
vscode-java-test|vsc|0.37.2023013002
vscode-maven|vsc|0.40.2023020103
vscode-spring-boot-dashboard|vsc|0.10.1
vscode-spring-initializr|vsc|0.11.2

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256198
vstes627cf:30244335
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
pythonsymbol12:30651887
6233i204:30659907

```

</details>

**Comment by meganrogge** [comment]
Doesn't repro on windows or mac ☹️

**Comment by meganrogge** [comment]
We shouldn't be focusing the last command via tab index when it's content editable

**Comment by jvesouza** [comment]
I can not reproduce using nvda.
However the content of the acessibility buffer is not presented by nvda. Instead of the content the following is presented:
No terminal content available for this session.

**Comment by meganrogge** [comment]
@jvesouza I misunderstood the issue - I can reproduce this when I use `contentEditable`. Will fix

**Comment by meganrogge** [comment]
> Instead of the content the following is presented: No terminal content available for this session.

This is what happens if you haven't run any commands yet or if shell integration is disabled

**Comment by jvesouza** [comment]
Weird because I moved the focus to the accessibility buffer after typing some commands like the ls command.
Also shell integration was not disabled by me and I was able to browse the accessibility buffer until yesterday.

This feature is extremely useful for me as using the terminal is part of the way I work.

**Comment by meganrogge** [comment]
@jvesouza can you repro it not working on windows w NVDA after you type commands? if so, i'm pretty sure shell integration isn't activating for you. what's the value of `terminal.integrated.windowsEnableConpty`?

**Comment by jvesouza** [comment]
I repeated the tests as requested and the result was the same.

"terminal.integrated.windowsEnableConpty": true

**Comment by meganrogge** [comment]
With `"terminal.integrated.windowsEnableConpty": true`, shell integration won't be activated, so it will not work. With that false, it should work.

**Comment by jvesouza** [comment]
I certainly must be doing something wrong because even if I set it to false I still get the same result.

Could this error be related to the fact that the language of my machine OS is different from English and the operating system is Windows 11?

**Comment by meganrogge** [comment]
I don't think either of those would cause this, but thanks for the info. What happens if you run the command `Terminal: Run recent command`? are there any items in the list?

**Comment by jvesouza** [comment]
> I don't think either of those would cause this, but thanks for the info. What happens if you run the command `Terminal: Run recent command`? are there any items in the list?

Although I ran several commands in powershell only one command is present in the list, and even then it's a command I didn't use in this session.
The only command that appears in the list is the pwd command, which I didn't run.

I don't know if it makes sense, but I suspect that the problem has to do with the version of powershell installed on my machine, version 5.1. What makes me think this is that I connected to a linux machine using remote ssh, opened a terminal and everything worked correctly.

**Comment by meganrogge** [comment]
Oh yep, that could be it. Is there a reason you can't update your powershell version?

**Comment by jvesouza** [comment]
> Oh yep, that could be it. Is there a reason you can't update your powershell version?

Two reasons!

1. I have little experience with Windows and I still don't know how to update.
2. The second is that the machine I'm using belongs to the company I work for and I don't have the admin password.
But I believe I can remove these two impediments and do the upgrade.

**Comment by jvesouza** [comment]
@meganrogge It seems that if in the accessibility buffer I move the cursor to a position before the last command, I still need to press the tab key twice to move focus to the terminal.

If for example the focus is on the last line of the accessibility buffer and I press tab, the focus is moved to the terminal. Now if the focus is on the first line of the accessibility buffer, I need to press the tab key twice.

**Comment by meganrogge** [comment]
@jvesouza are you on linux when that happens? I cannot repro on macOS

**Comment by meganrogge** [comment]
On the latest insider's:

On windows, when `terminal.integrated.accessibleBufferContentEditable` is false or auto, this is expected as tab index moves to the most recently run command. I can repro that happening. 

I cannot repro that happening when that is set to `on`.
